spec/outputs/elasticsearch_spec.rb

```diff
@@ -486,8 +486,13 @@ def mock_actions_with_response(*resp)
         .any_instance.stub(:bulk).and_return(*resp)
       LogStash::Outputs::Elasticsearch::Protocols::NodeClient
         .any_instance.stub(:bulk).and_return(*resp)
-      LogStash::Outputs::Elasticsearch::Protocols::TransportClient
-        .any_instance.stub(:bulk).and_return(*resp)
+    end
+
+    def unmock_actions()
```

Comment:
This might very necessary in here, usually stub scope is:

```
Doubles, stubs, and message expectations are all cleaned out after each
example. This ensures that each example can be run in isolation, and in any
order.
```

but I actually always found tricky to discover the right scope for mock of this kind. My best guess in here is that their scope is the context, so that is why in the way you want to use them you need to unstub them, and there is actually no shame on doing that way.

However we might try to refactor this test, or also including the plugin code, to now need this kind of constructs anymore. I for example wonder why do we need to mock the TransportClient, NodeClient and the HTTPClient if we might be only using one of them, for example.